File: my-app/src/types/menu.ts
export interface Menu {
  id: string | number;
  title: string;
  url: string;
  icon?: string;
  sortOrder: number;
  isActive: boolean;
  parentId?: string | number | null;
  parentTitle?: string | null;
  created?: string;
  children?: Menu[];
}

export interface CreateMenuInput {
  title: string;
  url: string;
  icon?: string;
  sortOrder: number;
  parentId?: string | number | null;
  isActive?: boolean;
}

File: API/Dms.Domain/Entities/Menu.cs
using Dms.Domain.Common;

namespace Dms.Domain.Entities
{
    public class Menu : BaseEntity
    {
        public string Title { get; set; } = string.Empty;
        public string Url { get; set; } = string.Empty;
        public string? Icon { get; set; }
        public int SortOrder { get; set; }
        public bool IsActive { get; set; } = true;
    }
}
